code/iaas/model/src/main/java/io/cattle/platform/core/dao/InstanceDao.java

```diff
@@ -19,4 +20,6 @@
      */
     List<? extends Service> findServicesFor(Instance instance);
 
+    List<? extends Instance> listNonRemovedInstances(Account account, boolean forService);
```

Comment:
@sonchang No subsequent PR, I've just wanted this method to handle both cases - return non-service and service instances.